modal/client.py

```diff
@@ -284,31 +282,33 @@ async def _reset_on_pid_change(self):
         if self._owner_pid and self._owner_pid != os.getpid():
             # not calling .close() since that would also interact with stale resources
             # just reset the internal state
-            self._channel = None
+            self._connection_pool = None
             self._stub = None
-            self._grpclib_stub = None
             self._owner_pid = None
 
             self.set_env_client(None)
             # TODO(elias): reset _cancellation_context in case ?
             await self._open()
 
-    async def _get_grpclib_method(self, method_name: str) -> Any:
+    # todo: decide if this should be moved into the connection pool
+    async def _get_grpclib_method(self, method_name: str, api_endpoint: str) -> Any:
         # safely get grcplib method that is bound to a valid channel
         # This prevents usage of stale methods across forks of processes
         await self._reset_on_pid_change()
-        return getattr(self._grpclib_stub, method_name)
+        stub = await self._connection_pool.get_grpclib_stub(api_endpoint)
+        return getattr(stub, method_name)
 
     @synchronizer.nowrap
     async def _call_unary(
         self,
         method_name: str,
         request: Any,
+        api_endpoint: str,
```

Comment:
Pretty confused by the terminology here. Do you mean hostname?

Endpoint to me is the url of a specific resource i.e. in this case a grpc method handler

Reply:
Good point. I renamed `api_endpoint` to `server_url`. Thanks!